File: app/src/test/java/com/utynote/components/search/FakeSearchView.java
package com.utynote.components.search;

import android.support.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

final class FakeSearchView implements SearchContract.View {

    @NonNull public final List<SearchViewModel.Data> results = new ArrayList<>();
    @NonNull public final List<SearchViewModel.Error> errors = new ArrayList<>();
    @NonNull public final List<SearchViewModel.Busy> busy = new ArrayList<>();


    @Override
    public void render(@NonNull SearchViewModel.Abstract model) {
        model.accept(new SearchViewModel.Visitor() {
            @Override
            public void visit(@NonNull SearchViewModel.Busy model) {
                busy.add(model);
            }

            @Override
            public void visit(@NonNull SearchViewModel.Error model) {
                errors.add(model);
            }

            @Override
            public void visit(@NonNull SearchViewModel.Data model) {
                results.add(model);
            }
        });
    }
}

File: app/src/test/java/com/utynote/components/search/FakeSearchView.kt
package com.utynote.components.search

import java.util.*

internal class FakeSearchView : SearchContract.View {

    val results: MutableList<SearchViewModel.Data> = ArrayList()
    val errors: MutableList<SearchViewModel.Error> = ArrayList()
    val busy: MutableList<SearchViewModel.Busy> = ArrayList()

    override fun render(model: SearchViewModel.Abstract) {
        when (model) {
            is SearchViewModel.Busy ->  busy.add(model)
            is SearchViewModel.Error ->  errors.add(model)
            is SearchViewModel.Data -> results.add(model)
        }
    }
}
